### ISSUE TITLE
Results of comparative experiment

### ISSUE THREAD

**Issue by Delete12137** [HaipengMing/SLM-AEA#1]
Thank you very much for your work!
I have successfully reproduced your work on RAFDB, but I would like to ask why your baseline (best) is about 2% higher than that in other papers when doing comparative experiments. And why is the reproduction of SCN much higher than the results reported in the original SCN paper? And what do best and last stand for in the table?In addition, there seems to be no description of the structure of baseline.
The following two tables show the comparison experiment of your paper and the experimental results of the original SCN paper.
![image](https://user-images.githubusercontent.com/105090442/228720469-7066aec2-ae6e-49c3-978b-31cf77ecd0b1.png)
![image](https://user-images.githubusercontent.com/105090442/228720497-4c7d982f-b57d-42c9-bafc-18bfddacc146.png)

**Comment by HaipengMing** [comment]
Thank you for your email!


We found that using different pre-trained model parameters has a significant impact on the results, and we speculate that the reason for better results should be due to the different pre-trained models used. Although the original SCN paper also pre-trained on the MS-Celeb-1M dataset, it seems that the pre-trained model we used is better. The code provided on Github uses a pre-trained model on ImageNet provided by Pytorch. If you want to achieve better results, you can use the pre-trained parameters provided by https://github.com/JDAI-CV/FaceX-Zoo/tree/main/addition_module/DMUE. The baseline network is resnet18, which is consistent with other papers. We trained for a total of 40 epochs, "best" refers to the highest accuracy on the test set among the 40 epochs, and "last" refers to the accuracy at the 40th epoch.



 Haipeng Ming
  ***@***.***
  



---- Replied Message ----
From  ***@***.***> Date  3/30/2023 11:26 To  ***@***.***> Cc  ***@***.***> Subject  [HaipengMing/SLM-AEA] Results of comparative experiment (Issue #1)  

Thank you very much for your work!
 I have successfully reproduced your work on RAFDB, but I would like to ask why your baseline (best) is about 2% higher than that in other papers when doing comparative experiments. And why is the reproduction of SCN much higher than the results reported in the original SCN paper? And what do best and last stand for in the table?In addition, there seems to be no description of the structure of baseline.
 The following two tables show the comparison experiment of your paper and the experimental results of the original SCN paper.

**Comment by Delete12137** [comment]
非常感谢你的回复！




| |
***@***.***
|
|
***@***.***
|




---- Replied Message ----
| From | ***@***.***> |
| Date | 03/30/2023 12:01 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [HaipengMing/SLM-AEA] Results of comparative experiment (Issue #1) |

Thank you for your email!


We found that using different pre-trained model parameters has a significant impact on the results, and we speculate that the reason for better results should be due to the different pre-trained models used. Although the original SCN paper also pre-trained on the MS-Celeb-1M dataset, it seems that the pre-trained model we used is better. The code provided on Github uses a pre-trained model on ImageNet provided by Pytorch. If you want to achieve better results, you can use the pre-trained parameters provided by https://github.com/JDAI-CV/FaceX-Zoo/tree/main/addition_module/DMUE. The baseline network is resnet18, which is consistent with other papers. We trained for a total of 40 epochs, "best" refers to the highest accuracy on the test set among the 40 epochs, and "last" refers to the accuracy at the 40th epoch.



Haipeng Ming
***@***.***




---- Replied Message ----
From ***@***.***> Date 3/30/2023 11:26 To ***@***.***> Cc ***@***.***> Subject [HaipengMing/SLM-AEA] Results of comparative experiment (Issue #1)

Thank you very much for your work!
I have successfully reproduced your work on RAFDB, but I would like to ask why your baseline (best) is about 2% higher than that in other papers when doing comparative experiments. And why is the reproduction of SCN much higher than the results reported in the original SCN paper? And what do best and last stand for in the table?In addition, there seems to be no description of the structure of baseline.
The following two tables show the comparison experiment of your paper and the experimental results of the original SCN paper.

**Comment by Delete12137** [comment]
你好，我尝试使用不同的预训练网络（在MS-Celeb-1M上）初始化baseline来进行实验观察不同的预训练模型对性能的影响，但是你推荐的DMUE那篇文章提供的似乎不只是resnet18的预训练模型，请问你是用DMUE提供的mv_epoch_17进行初始化的吗？

在复现的时候我修改了初始化模型的代码为
![image](https://user-images.githubusercontent.com/105090442/228999800-ccbf7cb9-0399-4f03-b5bc-b86532a824fe.png)

而无论是你源代码的初始化方法，还是我修改后的都无法成功初始化网络，似乎是因为网络结构不同的影响，即DMUE提供的不只是单纯的resnet18的模型参数。请问你是怎么初始化的。
![image](https://user-images.githubusercontent.com/105090442/229000472-273e4d0b-0938-430a-94b0-e5e0e112e335.png)

**Comment by HaipengMing** [comment]
Haipeng Ming
 
 ***@***.***
  



---- Replied Message ----
From  ***@***.***> Date  3/30/2023 12:46 To  ***@***.***> Cc  ***@***.***> ,
 ***@***.***> Subject  Re: [HaipengMing/SLM-AEA] Results of comparative experiment (Issue #1)  

非常感谢你的回复！




 | |
 ***@***.***
 |
 |
 ***@***.***
 |




 ---- Replied Message ----
 | From | ***@***.***> |
 | Date | 03/30/2023 12:01 |
 | To | ***@***.***> |
 | Cc | ***@***.***>***@***.***> |
 | Subject | Re: [HaipengMing/SLM-AEA] Results of comparative experiment (Issue #1) |

 Thank you for your email!


 We found that using different pre-trained model parameters has a significant impact on the results, and we speculate that the reason for better results should be due to the different pre-trained models used. Although the original SCN paper also pre-trained on the MS-Celeb-1M dataset, it seems that the pre-trained model we used is better. The code provided on Github uses a pre-trained model on ImageNet provided by Pytorch. If you want to achieve better results, you can use the pre-trained parameters provided by https://github.com/JDAI-CV/FaceX-Zoo/tree/main/addition_module/DMUE. The baseline network is resnet18, which is consistent with other papers. We trained for a total of 40 epochs, "best" refers to the highest accuracy on the test set among the 40 epochs, and "last" refers to the accuracy at the 40th epoch.



 Haipeng Ming
 ***@***.***




 ---- Replied Message ----
 From ***@***.***> Date 3/30/2023 11:26 To ***@***.***> Cc ***@***.***> Subject [HaipengMing/SLM-AEA] Results of comparative experiment (Issue #1)

 Thank you very much for your work!
 I have successfully reproduced your work on RAFDB, but I would like to ask why your baseline (best) is about 2% higher than that in other papers when doing comparative experiments. And why is the reproduction of SCN much higher than the results reported in the original SCN paper? And what do best and last stand for in the table?In addition, there seems to be no description of the structure of baseline.
 The following two tables show the comparison experiment of your paper and the experimental results of the original SCN paper.

**Comment by ming1753** [comment]
你好，你可以直接使用DMUE提供的ResNet18网络，可以使用pytorch提供的ResNet18。而在使用pytorch提供的ResNet18网络时，由于网络结构有细微差别，需要根据每层网络的”key“手动调整对应关系，借助module.state_dict().copy_()实现模型初始化。

**Comment by HaipengMing** [comment]
你好，你可以直接使用DMUE提供的ResNet18网络，也可以使用pytorch提供的ResNet18。而在使用pytorch提供的ResNet18网络时，由于网络结构有细微差别，需要根据每层网络的”key“手动调整对应关系，借助module.state_dict().copy_()实现模型初始化。

**Comment by Delete12137** [comment]
你好，请问直接使用DMUE提供的resnet18网络是什么意思呀，我也复现过DMUE的实验，但是他提供的模型就是他的整个框架呀。他是自己写的resnet18，并且把最后一个basicblock给修改为了多分枝，你的意思是在他的这个基础上修改得到resnet18吗？

抱歉，我还在学习过程中，对于很多东西还只是处于浅层的理解，只会用，改的话就没啥经验。

**Comment by HaipengMing** [comment]
这部分是DMUE写的ResNet：https://github.com/JDAI-CV/FaceX-Zoo/blob/main/addition_module/DMUE/models/resnet.py

**Comment by Delete12137** [comment]
好的，非常感谢！
